content/5_Plat/Segtree_Ext.mdx

```diff
@@ -111,6 +116,156 @@ int main() {
 }
 ```
 
+</CPPSection>
+</LanguageSection>
+
+<FocusProblem problem="walkSam2" />
+
+### Solution - Ordered Set
+
+First, we coordinate compress all the values and keep track of the frequency of each value in the array we build our segment tree on.
+We can easily answer the
+$\texttt{INSERT}$, $\texttt{DELETE}$, and $\texttt{COUNT}$ queries using
+the sum segment tree from the [PURS](/gold/PURS) module. Answering the
+$\texttt{K-TH}$ queries in $\mathcal{O}(\log{N})$ time requires us to use
+segment tree walking.
+
+Let $freq[i]$ equal the number of times $i$ occurs in our set.
+In our array of coordinate compressed values, we want to find the first
+indice $x$ such that $\sum_{i=0}^{x} freq[i]$ is $\geq K$.
+
+In the previous problem, we were walking on the prefix maximum of our array. Here, we are walking on the prefix sum of our array. The difference here is that if we know our answer is in $[l, r]$, then
+checking the maximum in our left and right childs is enough to know if we
+should walk left or right. But with sum, we also need to consider the range $[1, l)$ and how it might affect our answer.
+
+To handle keeping track of the prefix of $[1, l)$, we first set our prefix value to a neutral value, then join it with our left child's answer every time we walk right.
```

Comment:
wdym neutral value

Reply:
is it 0 in this case or

Reply:
ye

Reply:
should i clarify that in the explanation for this problem

Reply:
the prefix can be something else right
like say if it was multiplication it'd be 1?

Reply:
yeah if it was a segtree for multiplication on positive numbers then our "neutral value" would be 1

Reply:
yeah uh hb
"by neutral value we just mean the identity of the operation"
"so for addition it's 0, for mul it's 1, yada yada"

Reply:
done